File: documentation-examples-java/src/java/dev/typr/foundations/docs/core/ExecuteVoid.java
package dev.typr.foundations.docs.core;

import dev.typr.foundations.Fragment;
import dev.typr.foundations.Transactor;

@SuppressWarnings("unused")
public class ExecuteVoid {
  Transactor tx = null; // placeholder

  // start
  void applySchema() {
    tx.transactVoid(
        mc -> {
          mc.update(Fragment.of("CREATE TABLE users (id INT, name VARCHAR(100))"));
          mc.update(Fragment.of("CREATE INDEX idx_users_name ON users (name)"));
        });
  }
  // stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/core/ExecuteVoid.scala
package dev.typr.foundationssc.docs.core
import dev.typr.foundationssc.*
import dev.typr.foundationssc.Fragment.sql

@SuppressWarnings(Array("unused"))
object ExecuteVoid:
  var tx: Transactor = null // placeholder

  // start
  def applySchema(): Unit =
    sql"CREATE TABLE users (id INT, name VARCHAR(100))"
      .execute()
      .combine(sql"CREATE INDEX idx_users_name ON users (name)".execute())
      .voided()
      .transact(tx)
  // stop
